modules/odo-core-concepts.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/developer_cli_odo/understanding-odo.adoc
+
+:_content-type: CONCEPT
+[id="odo-core-concepts_{context}"]
+
+= odo core concepts
+
+`{odo-title}` abstracts Kubernetes concepts into a developer friendly terminology:
+
+Application::
+An _application_ in `odo` is a classic application, developed with a link:https://www.redhat.com/en/topics/cloud-native-apps[cloud-native approach], that is used to perform a particular task.
++
+Examples of applications: Online Video Streaming, Hotel Reservation System, Online Shopping.
+
+Component::
+A cloud-native application is a collection of small, independent, loosely coupled _components_. An `odo` component is a set of Kubernetes resources that  can run and be deployed separately.
++
+Examples of components: API Backend, Web Frontend, Payment Backend.
+
+Project::
+A _project_ contains your source code, tests, and libraries, organized in a single unit. Projects helps achieve multi-tenancy, allowing multiple teams to run multiple applications at the same time, on a single cluster.
+
+
+Context::
+The _context_ is a directory that contains the source code, tests, libraries and `odo` specific config files for a single component.
```

Comment:
```suggestion
The _context_ is a directory that contains the source code, tests, libraries, and odo-specific config files for a single component.
```

Reply:
Should we use "odo" or `{odo-title}` to refer to the app in general, and `odo` to refer to the related CLI command? When I look at the current topics, it's hard to determine what the right practice should be. The nearest comparison I get is by searching the repo for `oc`, which uses the full name to refer to the app and `oc` to refer to the command. For example "OpenShift CLI (`oc`)".

Reply:
In general, I would tend to hyphenate compounds, but in this instance I am not sure it is needed to clarify the intention, and it would look odd with odo rendered in one font  and the -specific in another.  I also worry that if we move to using a variable for odo, that a hyphenated "Red Hat odo-specific" would look bad.

So, I will simplify it as "odo config files" and avoid the issue entirely.